src/main/java/com/griddynamics/cloud/learning/dal/repositories/ArticleRepository.java

```diff
@@ -0,0 +1,114 @@
+            logger.log(Level.SEVERE, msg, e);
+            throw new AbortedQueryException(msg, e);
+        }
+    }
+
+    @Override
+    protected String getEntityTableName() {
+        return Article.ARTICLE_TABLE;
+    }
+
+    public DomainEntitiesDTO getAllSortedBy(final ArticleRequestParamsDTO params) {
+        String query = "SELECT * FROM article ORDER BY " + params.getFieldToOrderBy() + " " + params.getDirection();
+        return getEntitiesDTO(query, params);
+    }
+
+    public DomainEntitiesDTO getFreeOrPaidSortedBy(final ArticleRequestParamsDTO params) {
+        String query = "SELECT * FROM article WHERE free = " + params.getIsFree() +
+                " ORDER BY " + params.getFieldToOrderBy() + " " + params.getDirection();
+        return getEntitiesDTO(query, params);
+    }
+
+    public DomainEntitiesDTO getAllWithTagSortedBy(final ArticleRequestParamsDTO params) {
+        String query = "SELECT a.id, caption, description, content, price, author_id, currency, date, free " +
+                "FROM article AS a INNER JOIN tag_article AS ta ON a.id = ta.article_id " +
+                "INNER JOIN tag AS t ON ta.tag_id = t.id " +
+                "WHERE t.name = \'" + params.getTag() + "\' " +
+                "ORDER BY " + params.getFieldToOrderBy() + " " + params.getDirection();
+        return getEntitiesDTO(query, params);
+    }
+
+    public DomainEntitiesDTO getFreeOrPaidWithTagSortedBy(final ArticleRequestParamsDTO params) {
+        String query = "SELECT a.id, caption, description, content, price, author_id, currency, date, free " +
+                "FROM article AS a INNER JOIN tag_article AS ta ON a.id = ta.article_id " +
+                "INNER JOIN tag AS t ON ta.tag_id = t.id " +
+                "WHERE t.name = \'" + params.getTag() + "\' AND free = " + params.getIsFree() + " " +
+                "ORDER BY " + params.getFieldToOrderBy() + " " + params.getDirection();
+        return getEntitiesDTO(query, params);
+    }
+
+    public DomainEntitiesDTO getAllByCaptionSortedBy(final ArticleRequestParamsDTO params) {
+        String query = "SELECT * FROM article WHERE caption LIKE '%" + params.getCaptionPattern() +
+                "%' ORDER BY " + params.getFieldToOrderBy() + " " + params.getDirection();
+        return getEntitiesDTO(query, params);
+    }
+
+    public DomainEntitiesDTO getFreeOrPaidByCaptionSortedBy(final ArticleRequestParamsDTO params) {
+        String query = "SELECT * FROM article WHERE free = " + params.getIsFree() + " AND caption LIKE '%"
+                + params.getCaptionPattern() + "%' ORDER BY " + params.getFieldToOrderBy() + " " + params.getDirection();
+        return getEntitiesDTO(query, params);
+    }
+
+    public DomainEntitiesDTO getAllByCaptionWithTagSortedBy(final ArticleRequestParamsDTO params) {
+        String query = "SELECT a.id, caption, description, content, price, author_id, currency, date, free " +
```

Comment:
It's more convenient to specify a source table for all the fields in the select clause. Same for other requests